frontend/src/main/scala/bloop/logging/BspServerLogger.scala

```diff
@@ -91,7 +91,7 @@ final class BspServerLogger private (
     val problemPos = event.problem.position
     val problemSeverity = event.problem.severity
     val sourceFile = toOption(problemPos.sourceFile())
-    // TODO: Add code content
+    val code = toOption(event.problem.diagnosticCode()).map(_.code())
```

Comment:
@adpi2 I might need your insight in here like I did in sbt. I'm hitting on the same thing here as I am in Mill and this seems to always be empty when I'm testing with `3.2.1-RC1-bin-20220805-e560c2d-NIGHTLY`. It works find with sbt and is here, but when I'm trying with Mill or Bloop here I can't figure out why it's always empty.

Reply:
FYI I managed to reproduce it, and I am trying to understand why this is happening.

Reply:
So I tried to do inception: debugging Bloop by starting a debugger in Bloop and it just froze after a few steps... :laughing:

Reply:
Here is the :bug::
https://github.com/scalacenter/bloop/blob/a0ca82de1c0239e395045f2a73a1799b89f10ead/backend/src/main/scala/bloop/reporter/Reporter.scala#L140 

The diagnostic code must be copied here:
https://github.com/scalacenter/bloop/blob/a0ca82de1c0239e395045f2a73a1799b89f10ead/backend/src/main/scala/bloop/reporter/Reporter.scala#L97

Reply:
Bingo
```
[Trace - 08:11:45 pm] Received notification 'build/publishDiagnostics'
Params: {
  "textDocument": {
    "uri": "file:///Users/ckipp/Documents/scala-workspace/minimal/minimal3/src/Main.scala"
  },
  "buildTarget": {
    "uri": "file:/Users/ckipp/Documents/scala-workspace/minimal/minimal3/?id\u003dminimal3"
  },
  "diagnostics": [
    {
      "range": {
        "start": {
          "line": 4,
          "character": 18
        },
        "end": {
          "line": 4,
          "character": 19
        }
      },
      "severity": 1,
      "code": "7",
      "source": "bloop",
      "message": "Found:    (\u001b[31m3\u001b[0m : Int)\nRequired: String"
    }
  ],
  "reset": true
}
```
Amazing. So happy to see this. Thanks again!